substrate/validator-sets/pallet/src/lib.rs

```diff
@@ -326,20 +329,26 @@ pub mod pallet {
       while key_shares < u64::from(MAX_KEY_SHARES_PER_SET) {
         let Some((key, amount)) = iter.next() else { break };
 
-        InSet::<T>::set(Self::in_set_key(network, key), Some(()));
-        participants.push(key);
+        let these_key_shares = amount.0 / allocation_per_key_share;
+        InSet::<T>::set(Self::in_set_key(network, key), Some(these_key_shares));
+        participants.push((key, these_key_shares));
```

Comment:
Why was InSet/participants so modified?

Reply:
because we needed that info to pass the grandpa, babe pallets.

Reply:
Except we never read this info from InSet AFAICT, solely from Participants. That should remove the reason for InSet being modified.